vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/config/ingest/operator/OperatorConfig.kt

```diff
@@ -0,0 +1,129 @@
+package org.vitrivr.engine.core.config.ingest.operator
+
+import kotlinx.serialization.Serializable
+import org.vitrivr.engine.core.operators.ingest.Decoder
+import org.vitrivr.engine.core.operators.ingest.Transformer
+import org.vitrivr.engine.core.operators.ingest.Segmenter
+import org.vitrivr.engine.core.operators.ingest.Extractor
+import org.vitrivr.engine.core.operators.ingest.Exporter
+import org.vitrivr.engine.core.operators.ingest.Aggregator
+import org.vitrivr.engine.core.operators.ingest.Enumerator
+
+/**
+ * Configuration for ingestion operators, as defined in the [org.vitrivr.engine.core.operators.ingest] package.
+ * This is the definition of the operators, whereas the [OperationsConfig] defines the actual pipeline
+ */
+@Serializable(with = OperatorConfigSerializer::class)
+sealed class OperatorConfig {
+
+
+    /**
+     * The [OperatorType] of the ingestion operator described by this config.
+     */
+    abstract val type: OperatorType
+
+    /**
+     * Additional parameters, operator dependent.
+     */
+    abstract val parameters: Map<String, String>
+}
+
+sealed class FactoryBuildableOperatorConfig : OperatorConfig() {
+    /**
+     * The class name of the factory for the corresponding operator.
+     * See [org.vitrivr.engine.core.operators.ingest] for the available factories.
+     */
+    abstract val factory: String
+}
+
+/**
+ * Configuration for a [Decoder].
+ */
+@Serializable
+data class DecoderConfig(
+    override val factory: String,
+    override val parameters: Map<String, String> = mapOf()
+) : FactoryBuildableOperatorConfig() {
+    override val type = OperatorType.DECODER
+}
+
+/**
+ * Configuration for a [Enumerator]
+ */
+@Serializable
+data class EnumeratorConfig(

```

Comment:
As prevously, the media types are defined via parameter entry `mediaTypes`.
Arguably, this could be directly defined on the enumerator config itself.